### DISCUSSION TITLE
Problem accessing my Birdnet-Pi

### DISCUSSION THREAD

**Discussion by blogdemoi** [mcguirepr89/BirdNET-Pi#498] [Q&A]
Hi,

I set up my Birdnet-Pi two days ago and could access it on my local wifi network. Then, as I have a domain for it, I input my domain in the setting field. I also give my Pi a fixed IP in my network.

Now, although my Pi is connected to my wifi home network I can't access it on http://birdnet.local/ . As for accessing it through my domain name I can't either because I need the http://birdnet.local/ being accessible to configure the redirection in my router.

I can access my Pi via ssh though.

What could be wrong? Any idea?

Thank you.

**Comment by mcguirepr89** [comment]
Hi, @blogdemoi -- I'm sorry to hear you've had trouble sharing your installation.

May I ask how far along in [the sharing guide](https://github.com/mcguirepr89/BirdNET-Pi/wiki/Sharing-Your-BirdNET-Pi) you were able to get before encountering unexpected behavior?

**Comment by blogdemoi** [comment]
-Create DHCP Reservation done
EDIT -Configure Port Forwarding done (I think I do not need to do this as my Birdnet PI is connected to another PI via wifi that connects to a vpn, that's how I have a static public IP. To summarize, Birdnet Pi is connected to another Pi (connected to a vpn) which is connected to my home router). I hope it is not too confusing :-)
-My WAN Has A Static IP (Birdnetpi Pi is connected to my home server which has a public static IP through a vpn, I also gave the Birdnet Pi a static IP on the home network). I have my own domain and created a sub-domain for Birdnet Pi).
-Create DNS A record done
-Set an A record for your domain named "@" that points to your static public IP address. done
-Use a DNS lookup tool, ... BEFORE you move onto Step 2, else Step 2 will not work. done
-Input a Webpage password under "BirdNET-Pi Password" done
-Input your domain in the BirdNET-Pi URL under "Custom URL" done
-Click "Update Settings" and wait about 20 seconds and then go to your domain's URL done

I still have to configure my home router.

However I don't worry too much about this, but I worry because my Birdnet-PI is not locally accessible at http://birdnet.local/ since I did all these manipulations. I also had to assign a static IP adress to the Pi so it will always have the same adress on my home network, I did this in /etc/dhcpcd.conf , maybe this last manipulation interferes with Birdnet-Pi settings?

Anyway, I am re-installing everything now, but maybe you have suggestions about what I did before?

Thank you.

**Reply by mcguirepr89** [reply]
Hi, @blogdemoi 

>but maybe you have suggestions about what I did before?

Let's see . . .

>EDIT -Configure Port Forwarding done (I think I do not need to do this . . . 

Port forwarding will be 100% required since the internet will need to know how to reach ports 80 and port 443 on the BirdNET-Pi where the web server sits. Without telling the router (or VPN if it is routing traffic) exactly how requests that come in on ports 80 and 443 can reach the BirdNET-Pi, the BirdNET-Pi's web interface will have no way of responding.

>as my Birdnet PI is connected to another PI via wifi that connects to a vpn, that's how I have a static public IP. To summarize, Birdnet Pi is connected to another Pi (connected to a vpn) which is connected to my home router). I hope it is not too confusing :-)

Indeed this is confusing :)

- BirdNET-Pi connects to an access point that is present on another Raspberry Pi, (**RPI2**)
- **RPI2**'s access point connects you as a client on the VPN.
- The VPN is connected to your home router.

How is internet traffic meant to reach your BirdNET-Pi in this scenario? 

Is traffic meant to go:
1. VPN->BirdNET-Pi
2. Router->VPN->BirdNET-Pi
3. VPN->RPi2->BirdNET-Pi
4. Router->VPN->RPi2->BirdNET-Pi

Let me know what you had in mind since I do not understand the intended flow of traffic. Once I have an understanding of where the BirdNET-Pi is on your network, I will stand a better chance of helping.

My best regards,
Patrick

**Reply by blogdemoi** [reply]
Thank you for your help. Another question, if I enter a public url in the settings (my sub-domain I created), is my Birdnet-Pi page still accessible locally on my home network?

**Reply by mcguirepr89** [reply]
Yes -- by default, the address line for all Caddyfiles accepts any request on port 80 that reaches the device, whether via the hostname or directly through the IP address. This means, if the device is on the network and Caddy is running, it will respond to any method by which the request reaches port 80.

**Reply by blogdemoi** [reply]
And can I assign the Birdnet-Pi a fixed IP on my network without problems in /etc/dhcpcd.conf?

Thank you.

**Reply by blogdemoi** [reply]
Hum, it seems to work, can you check you can access my Birdnet-Pi https://birdnetpi.arnauld.org/ (it is still inside my house though). Thank you.

**Reply by ehpersonal38** [reply]
Looks good from here

**Reply by mcguirepr89** [reply]
>And can I assign the Birdnet-Pi a fixed IP on my network without problems in /etc/dhcpcd.conf?

Yes, this is required (the guide recommends a DHCP static lease/reservation) but any method of making the IP address static will function the same.

>Hum, it seems to work, can you check you can access my Birdnet-Pi https://birdnetpi.arnauld.org/ (it is still inside my house though). Thank you.

Looks good to me :)
![Screenshot from 2022-08-18 12-23-48](https://user-images.githubusercontent.com/60325264/185445781-6791a968-b091-4fd4-baa4-cd6a212130d7.png)

**Reply by blogdemoi** [reply]
Thank you for your help! 
I asked for a Birdweather ID and will now fine tune and read the documentatio/wiki and discussions here on Github. I plan to put the microphone, a 'Playstation eye" through the window this evening.

**Comment by blogdemoi** [comment]
Something else. I can have my Birdnet-Pi accessible with https://birdnetpi.arnauld.org/ . However I didn't fill in :

> When you update the URL below, the web server will reload, so be sure to wait at least 30 seconds and then go to your new URL.
> BirdNET-Pi URL:
> The BirdNET-Pi URL is how the main page will be reached. If you want your installation to respond to an IP address, place that here, but be sure to indicate "http://".
> Example for IP: http://192.168.0.109
> Example if you own your own domain: https://virginia.birdnetpi.com

But it is still accessible at https://birdnetpi.arnauld.org/

I just gave a password at the BirdNET-Pi Password section.

If I try to access the settings and live audio from https://birdnetpi.arnauld.org/ it asks for a username and password. However when I gave the credentials it doesn't work and prompt again for a username and psswd. Is it because I didn't fill in the BirdNET-Pi URL?
